src/gmp_backend.rs

```diff
@@ -4,7 +4,7 @@
 */
```

Comment:
I feel like I've already reviewed these changes elsewhere.  Are you porting a different PR to this repo?

Reply:
Yes, the safe prime algorithm is moved from axelarnetwork/rust-paillier#5. The main changes here is to extend the API to take in an RNG.